src/algebra/group/basic.lean

```diff
@@ -110,8 +119,7 @@ section left_cancel_monoid
 variables {M : Type u} [left_cancel_monoid M] {a b : M}
 
 @[simp, to_additive] lemma mul_eq_left_iff : a * b = a ↔ b = 1 :=
-calc a * b = a ↔ a * b = a * 1 : by rw mul_one
-           ... ↔ b = 1         : mul_left_cancel_iff
+⟨λ h, @mul_left_cancel _ _ a b 1 (by simp [h]), λ h, by simp [h]⟩
```

Comment:
Is there a reason to change these proofs? I find the `calc` ones more readable.

Reply:
I agree. I thought for more or less trivial lemmas like these, mathlib preferred short one-line term-mode proofs. I don't mind changing it back.

Reply:
Oh right. Yes, that's true in general, but `calc` proofs are already fairly close to term proofs (except where tactics are used in the steps), and using `simp` is "heavier" (in my mind) than the single `rw` that's in the `calc` proof. If we really wanted to golf these down to one line, the term equivalent of this proof would be:
```lean
(iff.refl _).trans  (by { rw [←mul_one a] {occs:=occurrences.pos [2]}, exact mul_left_cancel_iff })
```
This is mainly ugly due to specifying the precise location of the rewrite, but even if that weren't there, I think the `calc` comes out a bit nicer.

Reply:
Thanks for the detailed explanation. That makes sense. I changed back to calc.